### PR TITLE
[JENKINS-38550] Hide currently inapplicable LibraryRetrieverDescriptor’s

### PR DESCRIPTION
**Pull request by jglick**
[JENKINS-38550](https://issues.jenkins-ci.org/browse/JENKINS-38550)

@reviewbybees

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/workflow/libs/SCMRetriever.java [hunk 1]
```diff
@@ -72,7 +74,7 @@ public SCM getScm() {
             return "Legacy SCM";
         }
 
-        public static List<SCMDescriptor<?>> getSCMDescriptors() {
+        public List<SCMDescriptor<?>> getSCMDescriptors() {
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/workflow/libs/SCMRetriever.java]: 🐛 Binary compatibility change. Please keep [and deprecate] the original method and implement a new one if required

Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/libs/SCMRetriever.java]: Will not do; this was for internal use only. I will mark it as restricted however.

File: src/main/java/org/jenkinsci/plugins/workflow/libs/LibraryConfiguration.java [hunk 2]
```diff
@@ -116,6 +121,13 @@ public boolean isAllowVersionOverride() {
 
     @Extension public static class DescriptorImpl extends Descriptor<LibraryConfiguration> {
 
+        // TODO JENKINS-20020 ought to be unnecessary
+        public Collection<LibraryRetrieverDescriptor> getRetrieverDescriptors() {
```
Comment (daniel-beck) [inline src/main/java/org/jenkinsci/plugins/workflow/libs/LibraryConfiguration.java]: `@Restricted`?

**Comment by svanoort** [comment]
Implementation LGTM but waiting for the annotations to block external consumption of the APIs here (as others noted) once done I can happily bee it.

**Comment by svanoort** [comment]
also 🐝

**Comment by daniel-beck** [comment]
Here's a bee for you: :bee:

**Comment by oleg-nenashev** [comment]
Well, I'm approving a change since the maintainer feels strongly about the compat change. Not a bee

**Pull request merged**